raiden/transfer/utils.py

```diff
@@ -3,7 +3,32 @@
 from web3 import Web3
 
 from raiden.constants import EMPTY_HASH
+from raiden.storage.sqlite import SQLiteStorage
 from raiden.utils import typing
+from raiden.utils.serialization import serialize_bytes
+
+
+def get_latest_known_balance_proof(
+        storage: SQLiteStorage,
+        balance_hash: typing.BalanceHash,
+) -> typing.Optional['BalanceProofSignedState']:
+    """ Tries to find the balance proof with the provided balance hash
+    in either latest state change or latest event. """
+    state_change = storage.get_latest_state_change_by_data_field(
+        'balance_hash',
+        serialize_bytes(balance_hash),
+    )
+    if state_change:
+        return state_change.balance_proof
+
+    event = storage.get_latest_event_by_data_field(
```

Comment:
why are you also getting the events? these are not `BalanceProofSignedState` because these proofs are being sent by the current node.